service/controller/resource/clustersecret/openstack.go

```diff
@@ -0,0 +1,93 @@
+package clustersecret
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	capo "github.com/giantswarm/cluster-apps-operator/api/capo/v1alpha4"
+)
+
+type openStackClouds struct {
+	Clouds map[string]openStackCloudConfig `json:"clouds"`
+}
+
+type openStackCloudConfigAuth struct {
+	AuthURL        string `json:"auth_url"`
+	Username       string `json:"username"`
+	Password       string `json:"password"`
+	UserDomainName string `json:"user_domain_name"`
+	ProjectID      string `json:"project_id"`
+}
+
+type openStackCloudConfig struct {
+	Auth               openStackCloudConfigAuth `json:"auth"`
+	Verify             bool                     `json:"verify"`
+	RegionName         string                   `json:"region_name"`
+	Interface          string                   `json:"interface"`
+	IdentityAPIVersion int                      `json:"identity_api_version"`
+}
+
+func (r *Resource) generateOpenStackCloudConfig(ctx context.Context, cluster capo.OpenStackCluster) (map[string]interface{}, error) {
+	if cluster.Spec.IdentityRef == nil || cluster.Spec.IdentityRef.Name == "" || cluster.Spec.IdentityRef.Kind != "Secret" {
```

Comment:
@ericgraf I based this off of your MC setup script except I'm pulling the values from the cloud-config secret instead of environment variables. Let me know what you think.

Reply:
In the future, we might change some of these but for today this is great 👍